### ISSUE TITLE
Collection Tracker Overrides Deck Builder

### ISSUE THREAD

**Issue by VeXHarbinger** [Epix37/Hearthstone-Deck-Tracker#2341]
First off, amazing work.  I've downloaded your code set and dug in a bit myself.  I hope to be a helpful contributor one day soon.

As for the minor software imperfection I'm experiencing; I don't stay on top of adding items to the collection tracker (CT), at least as often as I should.  If you gain a new card and omit adding it to the collection tracker, after you add this unidentified card to the deck it doesn't alert you that the card isn't in the CT until you save the deck.  It then removes the card from the deck and alerts you, so if you're not thinking about it and go to play you get the cascading UI animations about missing cards and the card display continues to animate.  

As opposed to the current model, I'd like to propose that the on save alert about the card missing from the CT and auto-removing it that it offers you 3 options; either don't save and return to the deck builder, remove the card from your and save, or add the missing card(s) to the CT.  I feel this makes for an easier workflow (hopefully it's as easy to code as it is to explain) and reduces the number of steps currently required to rectify the situation.

**Comment by VeXHarbinger** [comment]
Additional Behavior note;  While in the deck builder I decided I needed a card and crafted it and added it to my deck in the deck builder.  Then, while still in edit mode, I opened the CT and added the new card to my collection.  However, when I saved the deck the alert was still displayed that my deck didn't match my known cards.  Now I'm pretty sure it did remove a card from my built deck on save, because when I re-opened it after saving (after seeing the card missing from my CT alert) there were only 29 cards in my deck, but the card it deleted was not the newly crafted card.  I say I'm pretty sure it did this because I don't recall seeing the incomplete deck alert prior to the unmatched CT collection.  However, I did get a bit of distracted in-between.  I'll try to remember to test this behavior next time I craft a card and will update this to confirm the exact behavior.
